src/LibraryManager/Providers/jsDelivr/JsDelivrCatalog.cs

```diff
@@ -0,0 +1,286 @@
+
+                    foreach (string packageName in packageNames)
+                    {
+                        CompletionItem completionItem = new CompletionItem
+                        {
+                            DisplayText = packageName,
+                            InsertionText = packageName
+                        };
+
+                        completions.Add(completionItem);
+                    }
+                }
+
+                // library version completion
+                else
+                {
+                    name = name[name.Length - 1] == '@' ? name.Remove(name.Length - 1) : name;
+
+                    completionSet.Start = name.Length + 1;
+                    completionSet.Length = version.Length;
+
+                    IEnumerable<string> versions;
+
+                    if (IsGitHub(name).Result)
+                    {
+                        versions = await GetGithubLibraryVersions(name);
+                    }
+                    else
+                    {
+                        JsDelivrLibraryGroup libGroup = new JsDelivrLibraryGroup(name);
+                        versions = await libGroup.GetLibraryIdsAsync(CancellationToken.None);
+                    }
+
+                    foreach (string libraryId in versions)
+                    {
+                        CompletionItem completionItem = new CompletionItem
+                        {
+                            DisplayText = libraryId,
+                            InsertionText = libraryId
+                        };
+
+                        completions.Add(completionItem);
+                    }
+                }
+
+                completionSet.Completions = completions;
+            }
+            catch (Exception ex)
+            {
+                _provider.HostInteraction.Logger.Log(ex.ToString(), LogLevel.Error);
+            }
+
+            return completionSet;
+        }
+
+        public async Task<IReadOnlyList<ILibraryGroup>> SearchAsync(string term, int maxHits, CancellationToken cancellationToken)
+        {
+            List<ILibraryGroup> libraryGroups = new List<ILibraryGroup>();
+
+            if (IsGitHub(term).Result)
+            {
+                return libraryGroups;
+            }
+
+            try
+            {
+                IEnumerable<string> packageNames = await Microsoft.Web.LibraryManager.Providers.Unpkg.NpmPackageSearch.GetPackageNamesAsync(term, CancellationToken.None);
+                libraryGroups = packageNames.Select(packageName => new JsDelivrLibraryGroup(packageName)).ToList<ILibraryGroup>();
+            }
+            catch (Exception ex)
+            {
+                _provider.HostInteraction.Logger.Log(ex.ToString(), LogLevel.Error);
+            }
+
+            return libraryGroups;
+        }
+
+        public async Task<IEnumerable<string>> GetGithubLibraryVersions(string name)
```

Comment:
Does this method need to be public? 
Seems to be used only in `JsDelivrCatalog` class.